test/cmd/builds.sh

```diff
@@ -13,55 +14,55 @@ project="$(oc project -q)"
-wait_for_command "oc get is ruby-20-centos7 | grep latest" $((60*TIME_SEC))
+wait_for_command "os::cmd::expect_success_and_text 'oc get is ruby-20-centos7' 'latest'" $((60*TIME_SEC))
 
 REAL_OUTPUT_TO=$(oc get bc/ruby-sample-build --template='{{ .spec.output.to.name }}')
-oc patch bc/ruby-sample-build -p '{"spec":{"output":{"to":{"name":"different:tag1"}}}}'
-oc get bc/ruby-sample-build --template='{{ .spec.output.to.name }}' | grep 'different'
-oc patch bc/ruby-sample-build -p "{\"spec\":{\"output\":{\"to\":{\"name\":\"${REAL_OUTPUT_TO}\"}}}}"
+os::cmd::expect_success "oc patch bc/ruby-sample-build -p '{\"spec\":{\"output\":{\"to\":{\"name\":\"different:tag1\"}}}}'"
+os::cmd::expect_success "oc get bc/ruby-sample-build --template='{{ .spec.output.to.name }}' | grep 'different'"
+os::cmd::expect_success "oc patch bc/ruby-sample-build -p '{\"spec\":{\"output\":{\"to\":{\"name\":\"${REAL_OUTPUT_TO}\"}}}}'"
 echo "patchAnonFields: ok"
 
-[ "$(oc describe buildConfigs ruby-sample-build | grep --text "Webhook GitHub" | grep -F "${url}/oapi/v1/namespaces/${project}/buildconfigs/ruby-sample-build/webhooks/secret101/github")" ]
-[ "$(oc describe buildConfigs ruby-sample-build | grep --text "Webhook Generic" | grep -F "${url}/oapi/v1/namespaces/${project}/buildconfigs/ruby-sample-build/webhooks/secret101/generic")" ]
-oc start-build --list-webhooks='all' ruby-sample-build
-[ "$(oc start-build --list-webhooks='all' bc/ruby-sample-build | grep --text "generic")" ]
-[ "$(oc start-build --list-webhooks='all' ruby-sample-build | grep --text "github")" ]
-[ "$(oc start-build --list-webhooks='github' ruby-sample-build | grep --text "secret101")" ]
-[ ! "$(oc start-build --list-webhooks='blah')" ]
+os::cmd::expect_success_and_text 'oc describe buildConfigs ruby-sample-build' "Webhook GitHub.+${url}/oapi/v1/namespaces/${project}/buildconfigs/ruby-sample-build/webhooks/secret101/github"
+os::cmd::expect_success_and_text 'oc describe buildConfigs ruby-sample-build' "Webhook Generic.+${url}/oapi/v1/namespaces/${project}/buildconfigs/ruby-sample-build/webhooks/secret101/generic"
+os::cmd::expect_success 'oc start-build --list-webhooks='all' ruby-sample-build'
+os::cmd::expect_success_and_text 'oc start-build --list-webhooks=all bc/ruby-sample-build' 'generic'
+os::cmd::expect_success_and_text 'oc start-build --list-webhooks=all ruby-sample-build' 'github'
+os::cmd::expect_success_and_text 'oc start-build --list-webhooks=github ruby-sample-build' 'secret101'
+os::cmd::expect_failure 'oc start-build --list-webhooks=blah'
 webhook=$(oc start-build --list-webhooks='generic' ruby-sample-build --api-version=v1 | head -n 1)
-oc start-build --from-webhook="${webhook}"
-oc get builds
-oc delete all -l build=docker
+# os::cmd::expect_success "oc start-build --from-webhook=${webhook}" # fails because the ruby imagestream is not found, but the start-build for the ruby sample build succeeds above for the same IS
```

Comment:
However, just above this, the `start-build` for the `ruby-sample-build` works and uses the same image stream.